### PR TITLE
Parent & al.

### LINKED ISSUE #12: Access to parent namespace
First: `tlaw` is awesome! I created https://github.com/marcandre/libraries_io with it and had a blast.

One frustration was that I couldn't find a way from an `APIPath` (`Namespace` or `Endpoint`) to access the parent `Namespace`. Is there one?

If not, would you consider adding one?

### PR DESCRIPTION
**Pull request by marcandre**
This PR fixes #12 by adding:
  * `parent`, `parents` & `lineage` as instance and singleton methods to `APIPath`
  * `Namespace.traverse`

It builds on [#15], but doesn't really depend on it.

### CODE REVIEW

File: lib/tlaw/has_parent.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    def parents
+      result = []
+      cursor = self
+      while (cursor = cursor.parent)
+        result << cursor
+      end
+      result
+    end
+
+    def lineage
```
Comment (zverok) [inline lib/tlaw/has_parent.rb]: I am not sure about the name, never met this word used in this context :)
Are there any other gems/Ruby project that provide the same API ("me and my parents") you aware of?.. What names do they prefer?

File: spec/tlaw/namespace_spec.rb [hunk 2]
```diff
@@ -91,6 +107,31 @@ module TLAW
...
+        end
+
+        it 'can yield all namespaces recursively' do
+          expect(namespace_class.traverse(:namespaces).to_a).to eq [
+            child_class,
+            grand_child_class
+          ]
+        end
+      end
     end
```
Comment (zverok) [inline spec/tlaw/namespace_spec.rb]: With [saharspec](https://github.com/zverok/saharspec) (already included) you can make it this way:
```ruby
subject { ->(*args) { namespace_class.traverse(*args).to_a } }

its_call { is_expected.to ret [...] }
its_call(:namespaces) { is_expected.to ret [...] }
its_call(:endpoints) { is_expected.to ret [...] }
# ...and also..
its_call(:garbage) { is_expected.to raise_error ArgumentError }
```
(`ret` is matcher name for "(its call) **returns**")

For me, this is more DRY and expresses "what's going on" better (params difference is not hidden in the middle of example, "what is test subject" also made obvious), but I will not insist, I heard different opinions on those tricks :)

**Review by zverok — APPROVED** [review]
Thank you!
I am willing to merge this :)
Please rebase. 
(I left a pair of comments, but neither is mandatory to react upon.)

File: lib/tlaw/has_parent.rb (hunk 1 above)
Comment (marcandre) [reply lib/tlaw/has_parent.rb]: Well, there's `Module#ancestors`, which includes the module itself... but we can't use that. It's not really mandatory. I've removed it from the PR, we can add it later if need be.

**Comment by marcandre** [comment]
Cool 😸 
Would it be possible to have a release?

**Comment by zverok** [comment]
Let's finish with all your open PRs discussion, and then we'll do it?

**Comment by marcandre** [comment]
Lol, personally, I make a release for any new feature or bug fix. I figure it costs nothing, and avoids waiting for a PR, then there's another open, etc... and time passes by.

**Comment by marcandre** [comment]
@zverok here's why I prefer to make frequent releases... It's been a month (since my request) the last release is more than a year old, and there are many differences between the current code and the released code. And there are still 3 open requests 😈

**Comment by marcandre** [comment]
(Give me the access and I'll do the releases for you!)

**Comment by zverok** [comment]
Yeah, you were right, I was wrong (expecting we'll "just finalize it" in 1-2 days). But currently [things are tough for me](https://github.com/molybdenum-99/tlaw/issues/4#issuecomment-439651030).
They'll be better.

**Pull request merged**